utilities/transactions/write_prepared_txn_db.h

```diff
@@ -606,24 +614,6 @@ class WritePreparedTxnDB : public PessimisticTransactionDB {
   std::shared_ptr<std::map<uint32_t, ColumnFamilyHandle*>> handle_map_;
 };
 
-class WritePreparedTxnReadCallback : public ReadCallback {
```

Comment:
i wonder why it is moved to write_prepared_txn.h?

Reply:
This is so that I can embed this object onto the WritePreparedTxn.

Strictly speaking, embedding only makes sense for write unprepared, since we need to access `txn->unprep_seq_` to correctly execute the read callback. However, I've done it for write prepared as well so that we can reuse more code.

Reply:
I think with overriding ::Get in WriteUnprepared there would be no need for mixing the callback in WritePrepared and WriteUnprepared.